src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,347 @@
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtils;
+
+/**
+ * This class is used internally in the build process to write a property file
+ * with short descriptions (the first sentences) of TokenTypes constants.
+ * Request: 724871
+ * For IDE plugins (like the eclipse plugin) it would be useful to have
+ * a programmatic access to the first sentence of the TokenType constants,
+ * so they can use them in their configuration gui.
+ * @author Pavel Bludov
+ */
+public final class JavadocPropertiesGenerator {
+
+    /**
+     * The command line option to specify the output file.
+     */
+    private static final String OPTION_DEST_FILE = "destfile";
+
+    /**
+     * The width of the CLI help option.
+     */
+    private static final int HELP_WIDTH = 100;
+
+    /**
+     * This regexp is used to extract the first sentence from the text.
+     * The end of the sentence is determined by the symbol "period", "exclamation mark" or
+     * "question mark", followed by a space or the end of the text.
+     */
+    private static final Pattern END_OF_SENTENCE_PATTERN = Pattern.compile("(.*?[.?!])(\\s|$)");
+
+    /**
+     * Don't create instance of this class, use the {@link #main(String[])} method instead.
+     */
+    private JavadocPropertiesGenerator() {
+    }
+
+    /**
+     * TokenTypes.properties generator entry point.
+     * @param args the command line arguments
+     * @throws CheckstyleException if parser or lexer failed or if there is an IO problem
+     * @throws ParseException if the command line can not be passed
+     **/
+    public static void main(String... args)
+            throws CheckstyleException, ParseException {
+        final CommandLine commandLine = parseCli(args);
+        if (commandLine.getArgList().size() == 1) {
+            final File inputFile = new File(commandLine.getArgList().get(0));
+            final File outputFile = new File(commandLine.getOptionValue(OPTION_DEST_FILE));
+            writePropertiesFile(inputFile, outputFile);
+        }
+        else {
+            printUsage();
+        }
+    }
+
+    /**
+     * Creates the .properties file from a .java file.
+     * @param inputFile .java file
+     * @param outputFile .properties file
+     * @throws CheckstyleException if a javadoc comment can not be parsed
+     * @noinspection IOResourceOpenedButNotSafelyClosed
```

Comment:
@pbludov , please update IDEA inspection config as I suggested, in this commit.
Please forget about usage of `noinspection` tag.  If you would like to use it please prepare very good reasons to make it.
We discussed this case before, I ma not ok to allow it.

Reply:
@pbludov , expected diff on config is:
```
$ git diff
diff --git a/config/intellij-idea-inspections.xml b/config/intellij-idea-inspections.xml
index dd10f2b..50e6d41 100644
--- a/config/intellij-idea-inspections.xml
+++ b/config/intellij-idea-inspections.xml
@@ -1017,10 +1017,10 @@ isolated classes and we cannot put them to separate package as it will affect us
     <inspection_tool class="IOResource" enabled="true" level="WARNING" enabled_by_default="true">
         <scope name="Production" level="WARNING" enabled="true">
             <option name="ignoredTypesString" value="java.io.ByteArrayOutputStream,java.io.ByteArrayInputStream,java.io.StringBufferInputStream,java.io.CharArrayWriter,java.io.CharArrayReader,java.io.StringWriter,java.io.StringReader" />
-            <option name="insideTryAllowed" value="false" />
+            <option name="insideTryAllowed" value="true" />
         </scope>
         <option name="ignoredTypesString" value="java.io.ByteArrayOutputStream,java.io.ByteArrayInputStream,java.io.StringBufferInputStream,java.io.CharArrayWriter,java.io.CharArrayReader,java.io.StringWriter,java.io.StringReader" />
-        <option name="insideTryAllowed" value="false" />
+        <option name="insideTryAllowed" value="true" />
     </inspection_tool>
     <!-- till we switch to jacoco we cannot use this as it conflicts with policy of 100% coverage -->
     <inspection_tool class="IfCanBeSwitch" enabled="false" level="ERROR" enabled_by_default="false" />

```

please let me know if it makes another problems.